src/util.rs

```diff
@@ -14,6 +14,26 @@ pub const BITS_PER_BYTE: u32 = 8;
 /// that are not valid in Rust ident
 const BLACKLIST_CHARS: &[char] = &['(', ')', '[', ']', '/', ' ', '-'];
 
+pub(crate) fn load_from(input: &str, config: &crate::util::Config) -> Result<Device> {
```

Comment:
I dont understand why this is living in util.rs instead if lib.rs

Reply:
I've tried to move `load_from` to `lib.rs` but get several errors looking like:

```
    = note: expected reference `&util::Config`
               found reference `&svd2rust::Config`
For more information about this error, try `rustc --explain E0308`.
```

Could you try it yourself?

Reply:
done, did some other changes as well.

the issue was that the util mod in main.rs and the util mod in lib.rs were separate things, and that means that the types were not equal.